### PR TITLE
Ports "Arcononomy: Personal Departmental Cargo Ordering" from tg

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
Ports [tgstation#53881](https://github.com/tgstation/tgstation/pull/53881) and [tgstation#59691](https://github.com/tgstation/tgstation/pull/59691)
Introduces a new app for modular tablets and laptops that allows people to request orders using their department budget
Heads and quartermasters can approve requests and call the cargo shuttle with the app but not send it back
All heads now start with a tablet that comes with the app preinstalled
`access_view` is a new var that determines what access someone needs to order with the app, so that people can't order whatever with their department's budget
The cargo console file has been renamed from console to orderconsole.dm to make it easier to identify
BZ and CO2 canister shipments are now atmos access because theres no reason for toxins to use bz and co2 is pretty lethal to release

## Why It's Good For The Game
Budgets are no longer cards heads keep hidden until the hos or something decides to load up the armory or the cmo decides to load up on meds only at the start of the round

## Changelog
:cl: lordScrubling, ArcaneMusic, RandomGamer123
add: cargo ordering app
add: heads start with a tablet with the app installed
tweak: bz and co2 canister shipments require atmos access
/:cl:

### CODE REVIEW

**Comment by BrumalLake** [comment]
Ah, i still need to make this scan ids that are on you since pcs dont come with an id module like tg
No idea why but it was flickering the shipments you could see when I did it before

**Comment by BrumalLake** [comment]
I still have to figure out why it replaces the approve/reject button with a 0 when you dont have permission
Not even tg has fixed it, so I don't really care as its just a small visual bug
image
image

**Comment by RDS88-beeman** [comment]
there is a reason for scientists to order BZ, and ironically it's for xeno, but all currently in rotation maps start with a can anyway so /shrug/

**Comment by BrumalLake** [comment]
> 
> 
> there is a reason for scientists to order BZ, and ironically it's for xeno, but all currently in rotation maps start with a can anyway so /shrug/

for xeno but requires toxins access 😳 useless even with xeno access yeah

**Comment by BrumalLake** [comment]
?rebuild tgui

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/budgetordering.dm:23]: Can't those two be static?

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/budgetordering.dm]:
```suggestion
	return EXPORT_CARGO
```

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/budgetordering.dm]:
```suggestion
	return id ? id : FALSE
```

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/budgetordering.dm]:
```suggestion
		var/obj/item/card/id/D = get_buyer_id(user)
```

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/budgetordering.dm:94]:
```suggestion
	data["loan"] = SSshuttle.shuttle_loan
```
?

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/budgetordering.dm:70]: You could build it once in ui static data

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (BrumalLake) [reply code/modules/modular_computers/file_system/programs/budgetordering.dm:94]: image
image
order console doing same thing argument as well

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (BrumalLake) [reply code/modules/modular_computers/file_system/programs/budgetordering.dm:70]: the order console does the same thing, don't want to make them different

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (BrumalLake) [reply code/modules/modular_computers/file_system/programs/budgetordering.dm:23]: the order console does the same thing, don't want to make them different

**Comment by ike709** [comment]
Okay so the maintainers approve of this and finally reviewed this in voicechat. However, rather than hiding packs you don't have access to it should instead just block ordering them.

**Comment by BrumalLake** [comment]
above addressed

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Seems to work as expected. The numbers are a bit off though. I believe the normal supply console ignores normal separation and doesn't include a space or comma. Can we mirror that here to keep it consistent? Otherwise good to go  

image

**Comment by lordScrubling** [comment]
normal console has separators as well in `formatMoney`

**Comment by BrumalLake** [comment]
image
all money in verdana now

File: code/modules/modular_computers/file_system/programs/budgetordering.dm
Comment (BrumalLake) [inline code/modules/modular_computers/file_system/programs/budgetordering.dm]:
```suggestion
```

**Pull request merged**
